datadog/dogstatsd/base.py

```diff
@@ -445,18 +456,24 @@ def _send_to_server(self, packet):
                 self.close_socket()
         except Exception as e:
             log.error("Unexpected error: %s", str(e))
-
         if self._telemetry:
             self.bytes_dropped += len(packet)
             self.packets_dropped += 1
 
     def _send_to_buffer(self, packet):
-        self.buffer.append(packet)
-        if len(self.buffer) >= self.max_buffer_size:
+        if self._should_flush(len(packet)):
             self._flush_buffer()
+        self.buffer.append(packet)
+        self._current_buffer_total_size += (len(packet) + 1)
```

Comment:
Can we had a quick comment that the `+ 1` is for the line break between payload ?

Reply:
Will do.